gateway-service/src/main/java/org/zowe/apiml/gateway/filters/SafIdtFilterFactory.java

```diff
@@ -36,28 +46,43 @@ public String getEndpointUrl(ServiceInstance instance) {
     }
 
     @Override
-    protected Mono<Void> processResponse(ServerWebExchange exchange, GatewayFilterChain chain, ZaasTokenResponse response) {
+    protected Mono<Void> processResponse(ServerWebExchange exchange, GatewayFilterChain chain, AuthorizationResponse<ZaasTokenResponse> tokenResponse) {
```

Comment:
I guess this method is not necessary anymore.

Reply:
But it depends on response from ZAAS (https://github.com/zowe/api-layer/blob/dd5a19bd60c8a4715f3748563325b47b9576e707/zaas-service/src/main/java/org/zowe/apiml/zaas/zaas/ZaasController.java#L127). The ZAAS defines the header to set and TokeFilterFactory (https://github.com/zowe/api-layer/blob/dd5a19bd60c8a4715f3748563325b47b9576e707/gateway-service/src/main/java/org/zowe/apiml/gateway/filters/TokenFilterFactory.java#L98)

I am not sure, but I assume a test mock wrong ZAAS response (without header name)

Reply:
Again, it's not getting to the controller in this test. The request is processed in the filter.

Reply:
Removed & fixed